query/parser/language.peg

```diff
@@ -127,25 +159,29 @@ add_one_pipe <-
   { p.addPipeExpression() }
   expression_annotation
 
-add_pipe <- (add_one_pipe) *
+add_pipe <- (add_one_pipe)*
 
 expression_atom <- expression_atom_raw expression_annotation
 
 expression_atom_raw <-
   expression_function /
   expression_metric /
   # #sub-expression
-  _ PAREN_OPEN expression_start _ PAREN_CLOSE /
+  (
+    _ PAREN_OPEN
+    (expression_start / &{ p.errorHere(position, `expected expression to follow "("`) })
+    (_ PAREN_CLOSE / &{ p.errorHere(position, `expected ")" to close "("`) })
+  ) /
   # constant scalar
   _ <DURATION> { p.addDurationNode(text) } /
-  _ <NUMBER> { p.addNumberNode(buffer[begin:end]) } /
-  _ STRING { p.addStringNode(unescapeLiteral(buffer[begin:end])) }
+  _ <NUMBER> { p.addNumberNode(text) } /
+  _ STRING { p.addStringNode(unescapeLiteral(text)) }
 
 expression_annotation_required <-
   _ "{"
   <[^}]*>
-  "}"
-  { p.addAnnotationExpression(buffer[begin:end]) }
+  ("}" / &{ p.errorHere(position, `expected "%CLOSEBRACE%" to close "%OPENBRACE%" opened for annotation`) })
```

Comment:
Oh- I actually need to handle these still.

The PEG parser generator doesn't actually handle Go's grammar in `{ }` blocks; it just assumes that they'll match up and when the last one closes, the block does too. This means they can't go in strings or comments without confusing go-peg a little.